### ISSUE TITLE
Code are constantly freezes

### ISSUE THREAD

**Issue by Josh65-2201** [microsoft/vscode#235556]
Type: <b>Bug</b>

When I'm trying to type in the code area itself and any popups (Auto complete, Copilot, code line information) will stop updating for a few seconds and is constantly happening. Scrolling will partially show the top and bottom of the area until all of it unfreezes again. This started happening with 1.95.2 I believe.

VS Code version: Code 1.95.3 (f1a4fb101478ce6ec82fe9627c43efbf9e98c813, 2024-11-13T14:50:04.152Z)
OS version: Windows_NT x64 10.0.26100
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 2600 Six-Core Processor             (12 x 3394)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.93GB (11.80GB free)|
|Process Argv|D:\\Joshs PC\\Games\\Servers\\Minecraft\\1.20.6 Paper - NeonCraftCity\\plugins\\Denizen\\scripts\\serverOther.dsc --crash-reporter-id 0ea9fbe9-16bc-407b-a35e-44a995b881ff|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (22)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-css-formatter|aes|1.0.2
vscode-m3u|af4|1.0.0
spellright|ban|3.0.140
denizenscript|den|1.4.7
copilot|Git|1.248.0
copilot-chat|Git|0.23.2024102903
vscode-autohotkey-plus-plus|mar|6.4.1
rainbow-csv|mec|3.13.0
vscode-html-format|moh|0.1.6
csharp|ms-|2.55.29
vscode-dotnet-runtime|ms-|2.2.3
debugpy|ms-|2024.12.0
python|ms-|2024.20.0
vscode-pylance|ms-|2024.12.1
remote-wsl|ms-|0.88.5
powershell|ms-|2024.4.0
vsliveshare|ms-|1.0.5941
indent-rainbow|ode|8.3.1
LiveServer|rit|5.7.9
HOCON|sab|0.0.1
vscode-gradle|vsc|3.16.4
volar|Vue|2.1.10

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
vscaac:30438847
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythonnoceb:30805159
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
dvdeprecation:31068756
dwnewjupytercf:31046870
2f103344:31071589
nativerepl1:31139838
pythonrstrctxt:31112756
nativeloc2:31192216
cf971741:31144450
iacca1:31171482
notype1cf:31157160
5fd0e150:31155592
dwcopilot:31170013
stablechunks:31184530

```

</details>

**Comment by alexr00** [comment]
Does the issue still happen if you disable all extensions?

**Comment by kanhao100** [comment]
I couldn't agree more
when I click on the windows taskbar, it will restore, I think this may be the windows system window manager or some other problem, this problem has been bothering me for a month, I can only restart vscode every time to succeed, it may occur 1 or 2 times a day
I have tested disabling all extensions and the problem still occurs

> Does the issue still happen if you disable all extensions?

**Comment by Josh65-2201** [comment]
Yes this is happening with extensions disabled. I also noticed that it happening with sidebar views like explorer active file selection colour

**Comment by alexr00** [comment]
Can you create a performance profile? https://github.com/microsoft/vscode/wiki/Performance-Issues#sluggish-ui

**Comment by kanhao100** [comment]
> Can you create a performance profile? https://github.com/microsoft/vscode/wiki/Performance-Issues#sluggish-ui

hello, we cannot create a performance profile, when I start the performance record and the freezes is disappeared.    I have tried it 3 times. 
by the way , it's also happened to Microsoft Edge 131.0.2903.86.

**Comment by Josh65-2201** [comment]
I was able to get a profile of it lagging
[Trace-20241213T234654.json.zip](https://github.com/user-attachments/files/18132936/Trace-20241213T234654.json.zip)

**Comment by deepak1556** [comment]
@Josh65-2201 do you have `"editor.experimentalEditContextEnabled"` enabled ? If so, can you disable and confirm if the issue repros.

Forced reflows in the attached trace point editcontext rendering, passing to you @aiday-mar

**Comment by Josh65-2201** [comment]
No it's off

**Comment by aiday-mar** [comment]
Hi from looking at your performance profile, it would seem you could do certain things to improve performance:

- disable showing suggestions, following image shows it is lagging for you:

<img width="682" alt="Image" src="https://github.com/user-attachments/assets/b55cbab0-780a-4c56-8f63-43d30040c61f" />

as well as here:

<img width="682" alt="Image" src="https://github.com/user-attachments/assets/ec76cf6f-36c8-4731-b92d-ee768d711455" />

- You could disable editor sticky scroll if you have it turned on (it can be a bit laggy)
- You mentioned you have "editor.experimentalEditContextEnabled" disabled. If you enable it do you see a performance improvement?

**Comment by kanhao100** [comment]
> > Can you create a performance profile? https://github.com/microsoft/vscode/wiki/Performance-Issues#sluggish-ui
> 
> hello, we cannot create a performance profile, when I start the performance record and the freezes is disappeared. I have tried it 3 times. by the way , it's also happened to Microsoft Edge 131.0.2903.86.

I found why in #167556 , thank you

**Comment by Josh65-2201** [comment]
Editor sticky scroll was already off and with `editor.experimentalEditContextEnabled` on it is frozen for less time but still happens just as often.

**Comment by aiday-mar** [comment]
pinging @jrieken about the suggestion showing performance

**Comment by jrieken** [comment]
@Josh65-2201 please provide a performance profile

**Comment by Josh65-2201** [comment]
I sent one here https://github.com/microsoft/vscode/issues/235556#issuecomment-2542558993

I'm not able to get another one right now, as when I try magically it stops.

**Comment by Josh65-2201** [comment]
This hasn't been happening for a while now. No idea what fixed it.
